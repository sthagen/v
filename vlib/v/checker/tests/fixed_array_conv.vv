arr := [2,3]!
mut p := voidptr(0)
p = arr
mut ip := &int(0)
ip = arr
_ = &int(arr)
