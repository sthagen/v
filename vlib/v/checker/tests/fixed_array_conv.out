vlib/v/checker/tests/fixed_array_conv.vv:3:3: error: mismatched types `voidptr` and `[2]int`
    1 | arr := [2,3]!
    2 | mut p := voidptr(0)
    3 | p = arr
      |   ^
    4 | mut ip := &int(0)
    5 | ip = arr
vlib/v/checker/tests/fixed_array_conv.vv:5:4: error: mismatched types `&int` and `[2]int`
    3 | p = arr
    4 | mut ip := &int(0)
    5 | ip = arr
      |    ^
    6 | _ = &int(arr)
    7 |
vlib/v/checker/tests/fixed_array_conv.vv:6:6: error: cannot cast a fixed array (use e.g. `&arr[0]` instead)
    4 | mut ip := &int(0)
    5 | ip = arr
    6 | _ = &int(arr)
      |      ~~~~~~~~
    7 | 
    8 | unsafe {
vlib/v/checker/tests/fixed_array_conv.vv:9:13: error: cannot use `[2]int` as `voidptr` in argument 1 to `memdup`
    7 | 
    8 | unsafe {
    9 |     _ = memdup(arr, 1)
      |                ~~~
   10 |     _ = tos(arr, 1)
   11 |     fn (p &int){}(arr)
vlib/v/checker/tests/fixed_array_conv.vv:10:10: error: cannot use `[2]int` as `byteptr` in argument 1 to `tos`
    8 | unsafe {
    9 |     _ = memdup(arr, 1)
   10 |     _ = tos(arr, 1)
      |             ~~~
   11 |     fn (p &int){}(arr)
   12 | }
vlib/v/checker/tests/fixed_array_conv.vv:11:16: error: cannot use `[2]int` as `&int` in argument 1 to `anon`
    9 |     _ = memdup(arr, 1)
   10 |     _ = tos(arr, 1)
   11 |     fn (p &int){}(arr)
      |                   ~~~
   12 | }
